### PR TITLE
dashboard/feature/handle_error_state

### PR DESCRIPTION
**Pull request by Asiasama710**
1-add all status code in app with exception error
2-hanlde error in taxi screen
3-handle error in resturant screen

### CODE REVIEW

**Review by hassanwasfy — APPROVED** [review]

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt [hunk 1]
```diff
@@ -182,16 +250,30 @@ class RestaurantScreenModel(
     }
 
     override fun onCancelCreateRestaurantClicked() {
-        updateState { it.copy(isNewRestaurantInfoDialogVisible = false, newRestaurantInfoUiState = NewRestaurantInfoUiState()) }
+        clearAddRestaurantInfo()
+        clearAddRestaurantErrorInfo()
+        updateState { it.copy(isNewRestaurantInfoDialogVisible = false) }
+    }
+
+    private fun clearAddRestaurantInfo() {
```
Comment (AzizaHelmy) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:258]: Refactor the name of this function

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt [hunk 2]
```diff
@@ -182,16 +250,30 @@ class RestaurantScreenModel(
...
+        updateState {
+            it.copy(
+                newRestaurantInfoUiState = it.newRestaurantInfoUiState.copy(
```
Comment (AzizaHelmy) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:261]: Refactor the name of this class "NewResturantInfoUiState"

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt [hunk 3]
```diff
@@ -291,21 +355,25 @@ class RestaurantScreenModel(
...
-            onSuccess = ::onCreateRestaurantSuccessfully,
-            onError = ::onError,
+            { manageRestaurant.createRestaurant(state.value.newRestaurantInfoUiState.toEntity()) },
+            ::onCreateRestaurantSuccessfully,
+            ::onError,
         )
     }
 
     private fun onCreateRestaurantSuccessfully(restaurant: Restaurant) {
+        clearAddRestaurantInfo()
```
Comment (AzizaHelmy) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:368]: Same refactor for the name

**Review by AzizaHelmy — APPROVED** [review]
Great work asuia👏🎉
3aaaaash alot

**Review by Youssef-Ramadan-Y — APPROVED** [review]

**Review by aya-seif — APPROVED** [review]

**Review by kamelkiko — APPROVED** [review]

**Pull request merged**